pyuvsim/antenna.py

```diff
@@ -73,13 +74,26 @@ def get_beam_jones(self, array, source_alt_az, frequency, reuse_spline=True,
         if freq_interp_kind is not None:
             array.beam_list[self.beam_id].freq_interp_kind = freq_interp_kind
 
-        array.beam_list[self.beam_id].interpolation_function = interpolation_function
+        if interpolation_function is not None:
+            array.beam_list[self.beam_id].interpolation_function = interpolation_function
+
+        spline_opts = None
+        if isinstance(array.beam_list, BeamList):
+            spline_opts = array.beam_list.spline_interp_opts
+
+        interp_kwargs = {'az_array' : source_az, 'za_array' : source_za,
+                         'freq_array' : freq, 'reuse_spline' : reuse_spline,
+                         'spline_opts' : spline_opts}
+
+        # For backwards compatibility.
+        try:
+            interp_data, interp_basis_vector = \
+                array.beam_list[self.beam_id].interp(**interp_kwargs)
+        except TypeError:
+            interp_kwargs.pop('spline_opts')
+            interp_data, interp_basis_vector = \
+                array.beam_list[self.beam_id].interp(**interp_kwargs)
```

Comment:
Is this for old versions of pyuvdata? This will result in the wrong thing being run. I think it should error with a helpful message about needing a newer version of pyuvdata if you want to use spline options.

Reply:
You're right. I've changed it to error when the interp call fails. However, to cover the except block we'd have to add a new test case using the older pyuvdata version. I've set the lines to be ignored by codecov.